[FSDP] Document optim_state_dict_config in method

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #102890
* #102692
* #102680
* #102672
* #102658
* __->__ #102657
* #102639
* #102637
* #102508

Per title

Differential Revision: [D46335318](https://our.internmc.facebook.com/intern/diff/D46335318/)

rohan-varma committed:
[FSDP] Document optim_state_dict_config in method

Per title

Differential Revision: [D46335318](https://our.internmc.facebook.com/intern/diff/D46335318/)

[ghstack-poisoned]

fegin reviewed (approved):
Nice

rohan-varma committed:
Update on "[FSDP] Document optim_state_dict_config in method"


Per title

Differential Revision: [D46335318](https://our.internmc.facebook.com/intern/diff/D46335318/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP] Document optim_state_dict_config in method"


Per title

Differential Revision: [D46335318](https://our.internmc.facebook.com/intern/diff/D46335318/)

[ghstack-poisoned]

awgu commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -749,8 +749,10 @@ def state_dict_type(
-                target ``state_dict_type``.
+            state_dict_config (Optional[StateDictConfig]): the model state_dict
+                configuration for the target ``state_dict_type``.
+            optim_state_dict_config: (Optional[OptimStateDictConfig]): the optimizer
```
```suggestion
            optim_state_dict_config (Optional[OptimStateDictConfig]): the optimizer
```

awgu commented on torch/distributed/fsdp/fully_sharded_data_parallel.py:
```diff
@@ -749,8 +749,10 @@ def state_dict_type(
             state_dict_type (StateDictType): the desired ``state_dict_type`` to set.
-            state_dict_config (Optional[StateDictConfig]): the configuration for the
-                target ``state_dict_type``.
+            state_dict_config (Optional[StateDictConfig]): the model state_dict
```
<img width="729" alt="Screenshot 2023-06-02 at 8 36 29 AM" src="https://github.com/pytorch/pytorch/assets/31054793/3be55d68-561c-465d-9546-0ffc50267cd5">

nit: I think we should use double backtick for "state_dict" or just call it "state dict" without the underscore.

rohan-varma committed:
Update on "[FSDP] Document optim_state_dict_config in method"


Per title

Differential Revision: [D46335318](https://our.internmc.facebook.com/intern/diff/D46335318/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP] Document optim_state_dict_config in method"


Per title

Differential Revision: [D46335318](https://our.internmc.facebook.com/intern/diff/D46335318/)

[ghstack-poisoned]
